File: 0509-fibonacci-number/0509-fibonacci-number.c


int fib(int n){

    if (n == 0 || n == 1){
            return n;
    }
    int prev2 = 0;
    int prev = 1;
        
    for (int i = 2; i <= n ; i++){
        int curri = prev2 + prev;
        prev2 = prev;
         prev = curri;
    }

    return prev;

}

File: 0509-fibonacci-number/0509-fibonacci-number.py
class Solution(object):
    def fib(self, n):
        if n == 0 or n == 1 :
            return n
        
        prev2 = 0
        prev = 1
        
        for i in range(2,n+1):
            curri = prev2 + prev
            prev2 = prev
            prev = curri
        

        return prev
        